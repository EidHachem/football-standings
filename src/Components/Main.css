.main-container {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: space-between;
  position: relative;
  min-height: 35vh;
}

.lower-container {
  display: flex;
  gap: 4em;
  z-index: 5;
  padding-bottom: 1em;
}

.main-container::after {
  content: '';
  position: absolute;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  background-image: url(../assets/football.png);
  background-size: cover;
  background-repeat: no-repeat;
  opacity: 0.8;
}

.leagues-logos {
  padding-left: 1em;
  display: grid;
  grid-template-columns: repeat(10, 1fr);
  grid-template-rows: auto auto;
  z-index: 3;
}

.total {
  z-index: 3;
  color: #fff;
  justify-self: flex-end;
  align-self: flex-end;
  margin-top: 15%;
  max-width: 7em;
}

.main-icon {
  justify-self: flex-end;
  align-self: center;
  margin-right: 1em;
}
