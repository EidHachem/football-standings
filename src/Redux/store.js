import { configureStore } from '@reduxjs/toolkit';
import leaguesSlice from './leaguesSlice/leaguesSlice';

const store = configureStore({
  reducer: {
    leagues: leaguesSlice,
  },
});

export default store;
