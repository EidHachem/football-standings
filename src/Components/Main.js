import React, { useEffect, useState } from 'react';
import { BiRightArrowCircle } from 'react-icons/bi';
import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import { FaSearch } from 'react-icons/fa';
import { IoChevronBackOutline } from 'react-icons/io5';
import { fetchLeagues } from '../Redux/leaguesSlice/leaguesSlice';
import Leagues from './Leagues';
import './Main.css';

const Main = () => {
  const leagues = useSelector((state) => state.leagues);
  const dispatch = useDispatch();
  const [search, setSearch] = useState(false);
  const [target, setTarget] = useState('');

  useEffect(() => {
    if (!leagues.length) dispatch(fetchLeagues());
  }, [dispatch, leagues.length]);

  return (
    <>
      <div className="nav-container">
        <Link to="/">
          <IoChevronBackOutline style={{ color: '#fff', opacity: '0', cursor: 'auto' }} />
        </Link>
        <p>Top teams</p>
        <div className="nav-icons">
          <FaSearch
            onClick={() => {
              setSearch(!search);
            }}
            style={{ cursor: 'pointer' }}
          />
        </div>
      </div>
      <div className="main-container">
        <p className="total">
          Total Available &nbsp; Leagues:
          {leagues.length}
        </p>
        <div className="lower-container">
          <div className="leagues-logos">
            {leagues.map((league) => (
              <div key={league.id}>
                <img src={league.logo} alt={league.name} className="logo" />
              </div>
            ))}
          </div>
          <Link to="/Leagues">
            <BiRightArrowCircle
              className="main-icon"
              style={{
                width: '25px',
                height: '25px',
                color: '#fff',
                border: 'none',
                zIndex: '4',
                opacity: '1',
              }}
            />
          </Link>
        </div>
      </div>
      <Leagues target={target} search={search} setTarget={setTarget} />
    </>
  );
};

export default Main;
