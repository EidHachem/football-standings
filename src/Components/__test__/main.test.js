import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../../Redux/store';
import Main from '../Main';

describe('Main test', () => {
  it('renders leagues', () => {
    const app = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <Main />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(app).toMatchSnapshot();
  });
});
