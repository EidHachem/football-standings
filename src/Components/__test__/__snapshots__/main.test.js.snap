// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main test renders leagues 1`] = `
Array [
  <div
    className="nav-container"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": "#fff",
            "opacity": "0",
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M328 112L184 256l144 144"
          fill="none"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="48"
        />
      </svg>
    </a>
    <p>
      Top teams
    </p>
    <div
      className="nav-icons"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 352 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M176 352c53.02 0 96-42.98 96-96V96c0-53.02-42.98-96-96-96S80 42.98 80 96v160c0 53.02 42.98 96 96 96zm160-160h-16c-8.84 0-16 7.16-16 16v48c0 74.8-64.49 134.82-140.79 127.38C96.71 376.89 48 317.11 48 250.3V208c0-8.84-7.16-16-16-16H16c-8.84 0-16 7.16-16 16v40.16c0 89.64 63.97 169.55 152 181.69V464H96c-8.84 0-16 7.16-16 16v16c0 8.84 7.16 16 16 16h160c8.84 0 16-7.16 16-16v-16c0-8.84-7.16-16-16-16h-56v-33.77C285.71 418.47 352 344.9 352 256v-48c0-8.84-7.16-16-16-16z"
        />
      </svg>
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M262.29 192.31a64 64 0 1057.4 57.4 64.13 64.13 0 00-57.4-57.4zM416.39 256a154.34 154.34 0 01-1.53 20.79l45.21 35.46a10.81 10.81 0 012.45 13.75l-42.77 74a10.81 10.81 0 01-13.14 4.59l-44.9-18.08a16.11 16.11 0 00-15.17 1.75A164.48 164.48 0 01325 400.8a15.94 15.94 0 00-8.82 12.14l-6.73 47.89a11.08 11.08 0 01-10.68 9.17h-85.54a11.11 11.11 0 01-10.69-8.87l-6.72-47.82a16.07 16.07 0 00-9-12.22 155.3 155.3 0 01-21.46-12.57 16 16 0 00-15.11-1.71l-44.89 18.07a10.81 10.81 0 01-13.14-4.58l-42.77-74a10.8 10.8 0 012.45-13.75l38.21-30a16.05 16.05 0 006-14.08c-.36-4.17-.58-8.33-.58-12.5s.21-8.27.58-12.35a16 16 0 00-6.07-13.94l-38.19-30A10.81 10.81 0 0149.48 186l42.77-74a10.81 10.81 0 0113.14-4.59l44.9 18.08a16.11 16.11 0 0015.17-1.75A164.48 164.48 0 01187 111.2a15.94 15.94 0 008.82-12.14l6.73-47.89A11.08 11.08 0 01213.23 42h85.54a11.11 11.11 0 0110.69 8.87l6.72 47.82a16.07 16.07 0 009 12.22 155.3 155.3 0 0121.46 12.57 16 16 0 0015.11 1.71l44.89-18.07a10.81 10.81 0 0113.14 4.58l42.77 74a10.8 10.8 0 01-2.45 13.75l-38.21 30a16.05 16.05 0 00-6.05 14.08c.33 4.14.55 8.3.55 12.47z"
          fill="none"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="32"
        />
      </svg>
    </div>
  </div>,
  <div
    className="main-container"
  >
    <p
      className="total"
    >
      Total Available Leagues:
      0
    </p>
    <div
      className="lower-container"
    >
      <div
        className="leagues-logos"
      />
      <a
        href="/Leagues"
        onClick={[Function]}
      >
        <svg
          className="main-icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "border": "none",
              "color": "#fff",
              "height": "25px",
              "opacity": "1",
              "width": "25px",
              "zIndex": "4",
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M11.999 1.993c-5.514.001-10 4.487-10 10.001s4.486 10 10.001 10c5.513 0 9.999-4.486 10-10 0-5.514-4.486-10-10.001-10.001zM12 19.994c-4.412 0-8.001-3.589-8.001-8s3.589-8 8-8.001C16.411 3.994 20 7.583 20 11.994c-.001 4.411-3.59 8-8 8z"
          />
          <path
            d="M12 10.994H8v2h4V16l4.005-4.005L12 7.991z"
          />
        </svg>
      </a>
    </div>
  </div>,
  <div
    className="grid-container"
  >
    <div>
      <h2>
        International leagues
      </h2>
    </div>
    <div
      className="leagues"
    />
  </div>,
]
`;
